src/runtime/pkg/katautils/config.go

```diff
@@ -1674,18 +1680,32 @@ func checkConfig(config oci.RuntimeConfig) error {
 // checkPCIeConfig ensures the PCIe configuration is valid.
 // Only allow one of the following settings for cold-plug:
 // no-port, root-port, switch-port
-func checkPCIeConfig(vfioPort hv.PCIePort, machineType string) error {
+func checkPCIeConfig(coldPlug config.PCIePort, hotPlug config.PCIePort, machineType string) error {
 	// Currently only QEMU q35 supports advanced PCIe topologies
 	// firecracker, dragonball do not have right now any PCIe support
 	if machineType != "q35" {
 		return nil
 	}
-	if vfioPort == hv.NoPort || vfioPort == hv.RootPort || vfioPort == hv.SwitchPort {
+
+	if coldPlug != config.NoPort && hotPlug != config.NoPort {
```

Comment:
@bergwolf Here is the check that both cannot be enabled simultaneously. Should we add the check in createPCIeTopology as well?

Reply:
Also added the check in the annotations, if cold-plug annotation reset hot-plug and vice-versa.